ot/partial.py

```diff
@@ -0,0 +1,1048 @@
+    >>> C1 = sp.spatial.distance.cdist(x, x)
+    >>> C2 = sp.spatial.distance.cdist(y, y)
+    >>> np.round(partial_gromov_wasserstein2(C1, C2, a, b),2)
+    1.69
+    >>> np.round(partial_gromov_wasserstein2(C1, C2, a, b, m=0.25),2)
+    0.0
+
+    References
+    ----------
+    ..  [29] Chapel, L., Alaya, M., Gasso, G. (2019). "Partial Gromov-
+        Wasserstein with Applications on Positive-Unlabeled Learning".
+        arXiv preprint arXiv:2002.08276.
+
+    """
+
+    partial_gw, log_gw = partial_gromov_wasserstein(C1, C2, p, q, m,
+                                                    nb_dummies, G0, thres,
+                                                    numItermax, tol, True,
+                                                    verbose, **kwargs)
+
+    log_gw['T'] = partial_gw
+
+    if log:
+        return log_gw['partial_gw_dist'], log_gw
+    else:
+        return log_gw['partial_gw_dist']
+
+
+def entropic_partial_wasserstein(a, b, M, reg, m=None, numItermax=1000,
+                                 stopThr=1e-100, verbose=False, log=False):
+    r"""
+    Solves the partial optimal transport problem
+    and returns the OT plan
+
+    The function considers the following problem:
+
+    .. math::
+        \gamma = arg\min_\gamma <\gamma,M>_F + reg\cdot\Omega(\gamma)
+
+        s.t. \gamma 1 \leq a \\
+             \gamma^T 1 \leq b \\
+             \gamma\geq 0 \\
+             1^T \gamma^T 1 = m \leq \min\{\|a\|_1, \|b\|_1\} \\
+
+    where :
+
+    - M is the metric cost matrix
+    - :math:`\Omega`  is the entropic regularization term
+        :math:`\Omega=\sum_{i,j} \gamma_{i,j}\log(\gamma_{i,j})`
+    - a and b are the sample weights
+    - m is the amount of mass to be transported
+
+    The formulation of the problem has been proposed in [3]_
+
+
+    Parameters
+    ----------
+    a : np.ndarray (dim_a,)
+        Unnormalized histogram of dimension dim_a
+    b : np.ndarray (dim_b,)
+        Unnormalized histograms of dimension dim_b
+    M : np.ndarray (dim_a, dim_b)
+        cost matrix
+    reg : float
+        Regularization term > 0
+    m : float, optional
+        Amount of mass to be transported
+    numItermax : int, optional
+        Max number of iterations
+    stopThr : float, optional
+        Stop threshold on error (>0)
+    verbose : bool, optional
+        Print information along iterations
+    log : bool, optional
+        record log if True
+
+
+    Returns
+    -------
+    gamma : (dim_a x dim_b) ndarray
+        Optimal transportation matrix for the given parameters
+    log : dict
+        log dictionary returned only if `log` is `True`
+
+
+    Examples
+    --------
+    >>> import ot
+    >>> a = [.1, .2]
+    >>> b = [.1, .1]
+    >>> M = [[0., 1.], [2., 3.]]
+    >>> np.round(entropic_partial_wasserstein(a, b, M, 1, 0.1), 2)
+    array([[0.06, 0.02],
+           [0.01, 0.  ]])
+
+    References
+    ----------
+    .. [3] Benamou, J. D., Carlier, G., Cuturi, M., Nenna, L., & Peyré, G.
```

Comment:
Add ref 29 here and discuss it in the doc. This is not the reference where partial entropic was proposed

Reply:
It has been defined in prop. 5 of ref 3